index.js

```diff
@@ -49,3 +49,21 @@ function compose (middleware) {
     }
   }
 }
+
+/**
+ * composes a wrapped middleware using middleware._wrappers
+ *
+ * @param {Function} middleware
+ * @return {Function} composed
+ * @api public
+ */
+
+function wrap (middleware) {
+  if (!middleware) return false
+  if (!middleware._wrappers) return middleware
+  if (!Array.isArray(middleware._wrappers)) throw new TypeError('._wrappers must be an array')
+  if (!middleware._wrappers.length) return middleware
+  const wrappers = middleware._wrappers
+  delete middleware._wrappers
+  return compose([...wrappers, middleware])
```

Comment:
It's not only an init-overhead - look at where the function is called - every time any middleware is executed, this function is called.

Also, I'm pretty sure you don't want to use compose internally here, instead use reduce. Look at my previous comment. That is, unless you want to change the wrapper style to something not really like a wrapper, more like a mw that runs before every mw.